packages/react-graphql/src/types.ts

```diff
@@ -58,4 +58,5 @@ export interface AsyncQueryComponentType<Data, Variables, DeepPartial>
   KeepFresh(
     props: VariableOptions<Variables> & {pollInterval?: number} & ConstantProps,
   ): React.ReactElement<{}>;
+  resolve(): Promise<DocumentNode<Data, Variables, DeepPartial> | null>;
```

Comment:
Shouldn't this always resolve to a real document node?

Reply:
not if the promise fail right?

Reply:
If it fails you should throw an error, there isn't a very good recovery case.